solutions/usaco-574.mdx

```diff
@@ -52,60 +52,95 @@ int main() {
 Instead of a knapsack solution, a BFS/DFS can be done to iterate though **all** attainable states and simply choose the one with maximum fullness. The scene can be thought of as a graph by defining each edge to be the transition from fullness $i$ to $j$. Each state is defined by fullness and whether Bessie has drank water. Since there are $2T$ states and each state is visited at most once, the time and space complexity of this solution is $\mathcal{O}(T)$. Below is a BFS implementation.
 
 <LanguageSection>
-
 <CPPSection>
 
 ```cpp
-#include <bits/stdc++.h>
-using namespace std;
-
-const int maxn = 5e6 + 5;
-bool pass[maxn][2];
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <set>
+#include <queue>
 
-struct node {
-	int val;
-	bool drunk;
-};
+using std::cout;
+using std::endl;
+using std::vector;
+using std::pair;
 
-int bfs(int t, int a, int b) {
-	int res = INT_MIN;
-	queue<node> q;
-	q.push({0, 0});
-
-	while (!q.empty()) {
-		int value = q.front().val;
-		bool drunk = q.front().drunk;
-		q.pop();
-		res = max(res, value);
-
-		if(value + a <= t && !pass[value + a][drunk]) {
-			q.push({value + a, drunk});
-			pass[value + a][drunk] = true;
-		}
-
-		if(value + b <= t && !pass[value + b][drunk]) {
-			q.push({value + b, drunk});
-			pass[value + b][drunk] = true;
+int main() {
+	std::ifstream read("feast.in");
+	int max_fullness;
+	int orange;
+	int lemon;
+	read >> max_fullness >> orange >> lemon;
+
+	std::set<pair<int, bool>> poss_fullness{{0, false}};
```

Comment:
explain more clearly that this is the visited array in the bfs

Reply:
this goes for the same in the python implementation

Reply:
fair enough